presto-main-base/src/main/java/com/facebook/presto/connector/system/AbstractPropertiesSystemTable.java

```diff
@@ -75,20 +79,45 @@ public final RecordCursor cursor(ConnectorTransactionHandle transactionHandle, C
     {
         TransactionId transactionId = ((GlobalSystemTransactionHandle) transactionHandle).getTransactionId();
 
+        /*
+        Get the list of catalogs. Catalog object will have CatalogContext which can help in retrieving the iceberg connector.
+        Catalog object has the Connector object as well.
+        Once you have the connector, get the deprecatedTableProperties and populate is_deprecated column.
+         */
         InMemoryRecordSet.Builder table = InMemoryRecordSet.builder(tableMetadata);
         Map<ConnectorId, Map<String, PropertyMetadata<?>>> connectorProperties = propertySupplier.get();
-        for (Entry<String, ConnectorId> entry : new TreeMap<>(transactionManager.getCatalogNames(transactionId)).entrySet()) {
-            String catalog = entry.getKey();
-            Map<String, PropertyMetadata<?>> properties = new TreeMap<>(connectorProperties.getOrDefault(entry.getValue(), ImmutableMap.of()));
-            for (PropertyMetadata<?> propertyMetadata : properties.values()) {
-                table.addRow(
-                        catalog,
-                        propertyMetadata.getName(),
-                        firstNonNull(propertyMetadata.getDefaultValue(), "").toString(),
-                        propertyMetadata.getSqlType().toString(),
-                        propertyMetadata.getDescription());
+        for (Catalog catalog : transactionManager.getCatalogs(transactionId)) {
+            String catalogName = catalog.getCatalogName();
+            ConnectorId connectorId = catalog.getConnectorId();
+            Map<String, PropertyMetadata<?>> properties = new TreeMap<>(connectorProperties.getOrDefault(connectorId, ImmutableMap.of()));
+            boolean isDeprecated = false;
+            if (catalog.getCatalogContext().getConnectorName().equalsIgnoreCase("iceberg")) {
+                Set<String> deprecatedPropertyNames = catalog.getConnector(connectorId).getDeprecatedTableProperties().stream().map(PropertyMetadata::getName).collect(Collectors.toSet());
+                for (PropertyMetadata<?> propertyMetadata : properties.values()) {
+                    isDeprecated = deprecatedPropertyNames.contains(propertyMetadata.getName());
+                    addTableRow(table, catalogName, isDeprecated, propertyMetadata);
+                }
+            }
+            else {
+                for (PropertyMetadata<?> propertyMetadata : properties.values()) {
+                    addTableRow(table, catalogName, isDeprecated, propertyMetadata);
+                }
```

Comment:
```suggestion
            for (PropertyMetadata<?> propertyMetadata : properties.values()) {
                boolean isDeprecated = deprecatedPropertyNames.contains(propertyMetadata.getName());
                addTableRow(table, catalogName, isDeprecated, propertyMetadata);
```
I believe we should avoid adding connector-specific logic to the code in `presto-main-base` module. As I see it, we can directly use the above connector-unrelated code to implement the same functionality. What do you think?